addons/sale/sale.py

```diff
@@ -824,6 +824,18 @@ def confirm_paid(self):
             order.message_post(body=_("Invoice %s paid") % (name))
         return res
 
+    @api.model
+    def _refund_cleanup_lines(self, lines):
+        result = super(AccountInvoice, self)._refund_cleanup_lines(lines=lines)
+        for line in lines:
+            values = {}
+            for name, field in line._fields.iteritems():
+                if name == 'sale_line_ids':
+                    values[name] = [(6, 0, line[name].ids)]
+            for r in result:
+                r[2].update(values)
```

Comment:
This is totally wrong.
It only work for single line. It will broke if it has multiple lines.

Here you assigning `sale_line_ids` of last line to all lines in result
so in result all records will have same `sale_line_ids`